penguin/web/routes.py

```
@@ -754,6 +758,73 @@ def _preview_text(value: Any, limit: int = 120) -> str:
+    explicit_title: str,
+    title_source: Optional[str],
+) -> Optional[str]:
+    """Return a reason to preserve the current title, or None if auto may run."""
+    if title_source == TITLE_SOURCE_MANUAL:
+        return "manual"
+
+    if title_source not in {"", TITLE_SOURCE_AUTO, None}:
+        return f"unknown_source:{title_source}"
+
+    if title_source in {"", None} and explicit_title:
+        if not _is_low_confidence_legacy_title(explicit_title):
```

Ruff (0.15.14)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)